engine/src/test/java/org/hibernate/validator/test/cfg/ConstraintMappingTest.java

```diff
@@ -145,7 +149,7 @@ public void testSingleConstraint() {
 
 		Set<ConstraintViolation<Marathon>> violations = validator.validate( new Marathon() );
 		assertNumberOfViolations( violations, 1 );
-		assertConstraintViolation( violations.iterator().next(), "may not be null" );
+		assertConstraintViolation( violations.iterator().next(), NotNull.class, "may not be null" );
```

Comment:
Yeah, to be honest, I did the minimum here to have something more consistent and a little better with as little effort as possible.

But I think we need something similar to what we do with containsOnlyPaths but more general:
```java
assertThat( constraintViolations ).containsOnlyViolations(
     violationWith().constraint( NotNull.class ).message( "may not be null" ).path( ... )
);
```

(and it would be possible to only test for the constraint type by using only `constraint()`)

It might just look like a cute thing but I think we have an important issue in our test suite: we test the constraint type and the path independently. We could have a mix-up between the constraint type and the path and it would not be reported as an error.

And it would also solve your issue with `assertNumberOfViolations`.